src/Interpreters/Context.cpp

```diff
@@ -2771,6 +2771,34 @@ void Context::shutdown()
     shared->shutdown();
 }
 
+std::vector<std::string> Context::listDisks()
+{
+    std::vector<std::string> disksNames;
+    std::lock_guard lock(shared->storage_policies_mutex);
+
+    for (auto & [disk_name, disk] : getDiskSelector(lock)->getDisksMap())
```

Comment:
getDiskSelector is a part of public interface of the context. You can do it inside disks tool code.